### ISSUE TITLE
Cross Platform symbols from ref-emit apis

### ISSUE THREAD

**Issue by KevinRansom** [dotnet/coreclr#1655]
Hi Guys,

when can we expect cross platform support PDB's from REF emit.  The F# compiler relies on ref-emit for Quotations and TypeProviders.  We aim to have cross platform compilation and debugging working for F# by the end of the year, it would be great if we can debug quotations and TP generated types too.

Thanks

Kevin

**Comment by jkotas** [comment]
The plan has been to use a new [portable debug info format](https://github.com/dotnet/roslyn/blob/master/docs/specs/PortablePdb-Metadata.md) for cross-plat. Talk to @tmat about the best way to move F# compiler over to it.

There are no plans to bring the Windows managed .pdbs cross-plat. Also, Reflection.Emit is limited for in-memory only in .NET Core. There are no plans to extend it to produce files on disk.

**Comment by tmat** [comment]
@jkotas I believe the question is rather whether in-memory assemblies emitted by Ref.Emit are debuggable on CoreCLR on Linux/Mac.

**Comment by tmat** [comment]
@KevinRansom Does F# save the Ref.Emit'd assemblies to disk for later loading?

**Comment by jkotas** [comment]
>  assemblies emitted by Ref.Emit are debuggable on CoreCLR on Linux/Mac

I do not think it is limited to Linux/Mac - the Ref.Emit surface on CoreCLR does not include the APIs to produce symbols, across platforms. The story for debuggable dynamically emitted assemblies in .NET Core has been to use `AssemblyLoadContext.LoadFromStream(System.IO.Stream assembly, System.IO.Stream assemblySymbols)`.

**Comment by tmat** [comment]
Oh, I see. So there is no way how to emit debuggable dynamic assembly thru Reflection.Emit? Looking at System.Reflection.Emit contract I don't even see Save - so assuming there is no way how to save an Ref.Emit'd assembly to disk/byte[]. So how can I use LoadFromStream?

**Comment by jkotas** [comment]
>  how can I use LoadFromStream?

Do not use System.Reflection.Emit...

**Comment by tmat** [comment]
@KevinRansom Why does F# use Reflection.Emit?
